gcloud-java-dns/src/test/java/com/google/gcloud/dns/DnsBatchTest.java

```diff
@@ -161,7 +173,30 @@ public void testListZonesWithOptions() {
     assertTrue(selector.contains(Dns.ZoneField.NAME.selector()));
     selector = (String) capturedOptions.getValue().get(ZONE_LIST_OPTIONS[3].rpcOption());
     assertEquals(DNS_NAME, selector);
-    // cannot test callback success without ManagedZonesListResponse which is final
+    // check the callback and result
+    // check the callback
+    ManagedZonesListResponse response = new ManagedZonesListResponse()
+        .setManagedZones(ZONE_LIST)
+        .setNextPageToken(PAGE_TOKEN);
+    RpcBatch.Callback<ManagedZonesListResponse> capturedCallback = callback.getValue();
+    EasyMock.verify(optionsMock);
+    EasyMock.reset(optionsMock);
+    EasyMock.expect(optionsMock.service()).andReturn(dns).times(ZONE_LIST.size());
+    EasyMock.replay(optionsMock);
+    capturedCallback.onSuccess(response);
+    Page<Zone> page = batchResult.get();
+    assertEquals(PAGE_TOKEN, page.nextPageCursor());
+    Iterator<Zone> iterator = page.values().iterator();
+    int resultSize = 0;
+    EasyMock.verify(dns);
+    EasyMock.reset(dns);
+    EasyMock.expect(dns.options()).andReturn(optionsMock).times(ZONE_LIST.size());
+    EasyMock.replay(dns);
+    while (iterator.hasNext()) {
+      assertEquals(ZONE_INFO.toPb(), iterator.next().toPb());
```

Comment:
Any particular reason for comparing pb objects rather than Zone objects? If you prefer like this I would suggest checking that the `.dns()` is populated correctly.